src/main/kotlin/com/petqua/application/product/review/ProductReviewService.kt

```diff
@@ -0,0 +1,39 @@
+package com.petqua.application.product.review
+
+import com.petqua.application.product.dto.ProductReviewReadQuery
+import com.petqua.application.product.dto.ProductReviewResponse
+import com.petqua.application.product.dto.ProductReviewsResponse
+import com.petqua.domain.product.dto.ProductReviewWithMemberResponse
+import com.petqua.domain.product.review.ProductReviewImageRepository
+import com.petqua.domain.product.review.ProductReviewRepository
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+@Transactional
+@Service
+class ProductReviewService(
+    private val productReviewRepository: ProductReviewRepository,
+    private val productReviewImageRepository: ProductReviewImageRepository,
+) {
+
+    @Transactional(readOnly = true)
+    fun readAll(query: ProductReviewReadQuery): ProductReviewsResponse {
+        val reviewsByCondition = productReviewRepository.findAllByCondition(
+            query.toCondition(),
+            query.toPaging(),
+        )
+        val imagesByReview = getImagesByReview(reviewsByCondition)
+        val responses = reviewsByCondition.map {
+            imagesByReview[it.id]?.let { images -> ProductReviewResponse(it, images) }
+                ?: ProductReviewResponse(it, emptyList())
+        }
```

Comment:
`?` 가 많으니 로직이 눈에 잘 안 들어오네요! 제가 이해한 바가 맞다면 다음과 같이 수정하면 어떨까요?

```suggestion
        val responses = reviewsByCondition.map {
            val images = imagesByReview[it.id] ?: emptyList()
            ProductReviewResponse(it, images)
        }
```

Reply:
images 따로 안 빼도 될 것 같기도 하구요!
```suggestion
val responses = reviewsByCondition.map {
            ProductReviewResponse(it, imagesByReview[it.id] ?: emptyList()) 
        }
```

Reply:
오.. 맞습니다👍👍